### ISSUE TITLE
Colony Creation Accessibility 

### ISSUE THREAD

**Issue by naynovi** [JoinColony/colonyCDapp#1919]
### tl:dr

We want to give users a choice on their colony creation URL. We should implement a few safeguards to protect our domain authority.

### Description

Users may choose nearly any URL they wish in the Colony creation process. Colony, nor the Users themselves, cannot change these later. We should prevent some of the worst terms being used to preserve our site authority and limit opportunities for malicious actions/spoofing attempts. 

### SEO reliability
Google downgrades site discoverability when domains are connected to URLs that look suspicious or may lead to harmful content. Robots have a basic vocabulary and are scared easily. Poor user-created URLs can impact the visibility and discovery of the application as a whole, which we will be unable to improve later. If we introduce a few filters it will prevent some of the worst terms being used and attached to our sitemap.

*Bad Words* Google doesn't like bad words ![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/92b9e336-b7f6-4e65-af11-8938ef1dd687)

*Underscores* 
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/ca951af4-1fc8-407c-8946-9199da9f85f4)
https://developers.google.com/search/docs/crawling-indexing/url-structure

*Reserved text*
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/c4a36b4e-9d48-49e0-b18b-1985274f0148)
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/ce61a27c-f210-4ce7-b482-a78eff93c984)
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/3735e4c5-3911-4728-931f-25ccd6f5c55b)
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/c1a2be99-9a2f-497f-ac45-8cf9c815085d)
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/0844b4cd-537d-4abb-93ca-74520c192ee7)

^these might not be issues but might be good to exclude

### Spoofing: 

An issue that has been raised by users previously
![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/3333010b-bb0a-47e1-b20f-8b98252d2985)

![image](https://github.com/JoinColony/colonyCDapp/assets/23479095/c954e9b9-afc9-4439-bb42-b54cb33adb9a)

## How we could do it

- Bad words/Underscores: Add the following terms (https://github.com/coffee-and-fun/google-profanity-words/tree/main ) as well as _ to colony name regex 

So when 

```
const isNameTaken = createYupTestFromQuery({
  query: CheckColonyNameExistsDocument,
  circuitBreaker: isValidName,
});
```
is run, it returns `'error.colonyURL': 'This is not a valid colony URL'`

https://github.com/JoinColony/colonyCDapp/blob/0ef19a797b5f813cd56196b2bc09b4f5474eeb28/src/components/common/Onboarding/wizardSteps/CreateColony/validation.ts#L132 

- Spoofing: As above, add list of Country names (maybe from here https://github.com/annexare/Countries) to colony name regex to at prevent delicious opportunity to impersonate a government authority. Maybe include some of the capital cities or prominent locations in that list as well. 

- Reserved text: maybe we have a list of things we'd like to not allow, 404 and index seem like obvious candidates.

- I'm not sure how we can get around corporate impersonation unless we add a similar list of names. 

### This issue is complete when...

- [ ] Users receive an error notice when they enter prohibited text in the URL generation field.

**Comment by rdig** [comment]
Good ideas, however a few pointers:
- Don't suggest packages or services, that's not your job to research, plus most likely we'll not be using those anyway. Instead, detail the problem that you want solved, it's up to us to figure how and using what
- Don't go digging in the code to suggest implementation details, again, it's not your job. Instead, again, tell us what problem you want solved, and we'll work with you to solve it
- The spec for this was already established prior to the implementation of names, both for colonies and users. If you want to improve the spec fine, but this sounds more like you thinking there is/was no spec to being with
- There already exists a blacklist of routes that are not allowed, to wither be accessed or created, which is basically what you suggest here: https://github.com/JoinColony/colonyCDapp/blob/4aab7d9e3e602c7c0c63494f1ee7cc78af5d32f2/src/routes/routeConstants.ts#L71-L206
- Some of the names you suggest are very valid as a colony or user name. While childish sure, I see no reason why we'd restrict the use of `butt` or `unitedstates` as a colony name, it's really not our place to enforce that

**Comment by naynovi** [comment]
And the reason why I added the repository is because it had identified those words for you.
